8.x/theme-development/helpers.md

```diff
@@ -2,4 +2,2097 @@
+    {
+        return $article->directory->articles()->latest()->except($article->id)->take($count)->get();
+    }
+}
+```
+
+
+### get_experts()
+This function will be deprecated in the next version.
+
+```php
+if (!function_exists('get_experts')) {
+     /**
+     * @deprecated
+     */
+    function get_experts(int $count = 4)
+    {
+        return SystemUser::where('is_expert', true)->take($count)->get();
+    }
+}
+```
+
+
+### recaptcha_enabled()
+This function determines whether Recaptcha is enabled on the system.
+
+```php
+if (!function_exists('recaptcha_enabled')) {
+    /**
+     * @return bool
+     */
+    function recaptcha_enabled(): bool
+    {
+        return strpos(env("TEMPORARILY_DISABLED_RULES", ""), "g-recaptcha-response") === false;
+    }
+}
+```
+
+
+### get_same_models_products()
+Takes a product from the input and returns similar products. For example: Different sizes of a shoe are similar products.
+
+```php
+if (!function_exists('get_same_models_products')) {
+    /**
+     * @param $product
+     * @return array
+     */
+    function get_same_models_products($product): array
+    {
+        $products = Product::models($product, false)
+            ->with('productStructure', 'images', 'rates')
+            ->get();
+        return json_decode($products);
+    }
+}
+```
+
+
+### check_cart()
+Checks if the product is in the cart. It is suggested not to use this function.
+
+```php
+if (!function_exists('check_cart')) {
+    function check_cart($product_id): bool
+    {
+        $customer = get_customer_user();
+        if ($customer !== false) {
+            return $customer->cartRows()->where("product_id", $product_id)->count() > 0;
+        } else {
+            $cart_rows = get_local_cart();
+            foreach ($cart_rows as $cart_row) {
+                if ($cart_row->product_id === $product_id) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+}
+```
+
+
+### get_cart_information()
+This function returns cartRows data related to a product_id.
+
+```php
+if (!function_exists('get_cart_information')) {
+    function get_cart_information($product_id)
+    {
+        $customer = get_customer_user();
+        $selected_row = null;
+        if ($customer !== false) {
+            $selected_row = $customer->cartRows()->with('product')->where("product_id", $product_id)->first();
+        } else {
+            $cart_rows = get_local_cart();
+            foreach ($cart_rows as $cart_row) {
+                if ($cart_row->product_id === $product_id) {
+                    $product = Product::find($product_id);
+                    $selected_row = $cart_row;
+                    $selected_row->product = $product;
+                    $selected_row->id = $product->id;
+                }
+            }
+        }
+        return $selected_row;
+    }
+}
+```
+
+
+### get_cart()
+-------------------------------------------
```

Comment:
This function returns the structured cart of the current customer whether he is logged in or not.